code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -654,16 +662,14 @@
 			update_icons()
 
 /mob/living/simple_animal/hostile/proc/set_targets_from(atom/target_from)
-	if(targets_from)
-		UnregisterSignal(targets_from, COMSIG_PARENT_QDELETING)
-	targets_from = target_from
-	if(targets_from)
-		RegisterSignal(targets_from, COMSIG_PARENT_QDELETING, .proc/handle_targets_from_del)
+	targets_from = WEAKREF(target_from)
 
-/mob/living/simple_animal/hostile/proc/handle_targets_from_del(datum/source)
-	SIGNAL_HANDLER
-	if(targets_from != src)
+/mob/living/simple_animal/hostile/proc/get_targets_from()
```

Comment:
AI targeting is pretty hot, would you mind adding a guardian var for this so there's no proccall cost here ?

Reply:
what would that look like?

Reply:
Something similar to ATMOS_PASS_YES/NO. 
`#DEFINE GET_TFM(X) X.targets_from_guard_var ? src : get_targets_from())`

Reply:
alright, sure